src/WritableMemoryStream.ts

```diff
@@ -0,0 +1,39 @@
+const util = require('util');
+const stream = require('stream');
+const Writable = stream.Writable;
+
+/**
+ * Writable Memory stream class
+ */
+export default class WritableMemoryStream extends Writable {
+  /** variable to store stream data */
+  private memStore: Buffer;
+
+  /**
+   * Constructor to initialize the memory stream
+   * @param key Unique key to identify data.
+   * @param options Other options for the writable stream.
+   */
+  public constructor (options?: any) {
+    super();
+
+    Writable.call(this, options);
+    this.memStore = Buffer.from('');
+  }
+
+  /**
+   * Return stored inmem stream data.
+   */
+  public fetchData (): Buffer {
+    return this.memStore;
+  }
+}
+
+/**
+ * Overwrite write method of the parent class to write to inmem store.
+ */
+WritableMemoryStream.prototype._write = function (chunk: string, enc: string, cb) {
+  let buffer = (Buffer.isBuffer(chunk) ? chunk : Buffer.from(chunk, enc));
+  this.memStore = Buffer.concat([this.memStore, buffer]);
```

Comment:
Design: I am concerned about the efficiency of this implementation, as this looks like memory reallocations and copying will keep occurring for as many times `_write` is invoked. An alternative approach is to store the buffer in an array and only invoke `concat` when `fetchData()` is invoked.

Reply:
i though about this, but since i dont know the size beforehand i couldnt declare an array, and push the elements. Please lmk if you can suggest an alternate